src/test/java/bio/terra/common/iam/AuthenticatedUserRequestTest.java

```diff
@@ -119,4 +125,69 @@ public void testVectors() throws JsonProcessingException {
             .setToken("0123.456-789AbCd")
             .build());
   }
+
+  @Test
+  public void testProtoInFlightMap() {
+    AuthenticatedUserRequest request =
+        AuthenticatedUserRequest.builder()
+            .setEmail(EMAIL_ADDRESS)
+            .setSubjectId(SUBJECT_ID)
+            .setToken(TOKEN)
+            .build();
+
+    FlightMap inMap = request.putIn(new FlightMap(), "foo");
+    FlightMap outMap = new FlightMap();
+    outMap.fromJson(inMap.toJson());
+
+    AuthenticatedUserRequest retrieved = AuthenticatedUserRequest.getFrom(outMap, "foo");
+    assertEquals(request, retrieved);
+  }
+
+  /**
+   * Show backwards/forward compatibility of protobuf messages. This does not relate to
+   * AuthenticatedUserRequests directly.
+   *
+   * <p>See
+   * https://developers.google.com/protocol-buffers/docs/javatutorial#extending-a-protocol-buffer
+   */
+  @Test
+  public void protoCompatibility() {
+    AuthenticatedUserRequestModel messageV1 =
+        AuthenticatedUserRequestModel.newBuilder()
+            .setEmail(EMAIL_ADDRESS)
+            .setSubjectId(SUBJECT_ID)
+            .setToken(TOKEN)
+            .build();
+
+    AuthenticatedUserRequestModelV2 messageV2 =
+        AuthenticatedUserRequestModelV2.newBuilder()
+            .setEmail(EMAIL_ADDRESS)
+            .setSubjectId(SUBJECT_ID)
+            .setRenamedToken(TOKEN)
+            .setAdditionalField("foo")
+            .build();
+
+    FlightMap inMap = new FlightMap();
+    putMessage(inMap, "v1", messageV1);
+    putMessage(inMap, "v2", messageV2);
+    FlightMap outMap = new FlightMap();
+    outMap.fromJson(inMap.toJson());
```

Comment:
I'd be interested to see the pretty-printed output of `fromJson()` to assess human readability for debug.

Reply:
The binary format is not aimed at human readability: https://developers.google.com/protocol-buffers/docs/encoding
protobufs do have a pretty print text format encoding (& a JSON encoding), but they have different schema evolution properties (e.g. the field name is used, instead of the field tag number).
Adding some logger.info with comments for visibility.